test/e2e-helm/e2e_helm_suite_test.go

```diff
@@ -0,0 +1,144 @@
+	isOLMManagedBySuite = true
+	// Kubectx staore the k8s context from where the tests are running
+	Kubectx string
+)
+
+// BeforeSuite run before any specs are run to perform the required actions for all e2e Helm tests.
+var _ = BeforeSuite(func(done Done) {
+	var err error
+
+	By("Creating a new test context")
+	tc, err = testutils.NewTestContext("GO111MODULE=on")
+	Expect(err).NotTo(HaveOccurred())
+	Expect(tc.Prepare()).To(Succeed())
+
+	By("Checking the cluster type")
+	Kubectx, err = tc.Kubectl.Command("config", "current-context")
+	Expect(err).Should(Succeed())
+
+	By("Checking API resources applied on Cluster")
+	output, err := tc.Kubectl.Command("api-resources")
+	Expect(err).NotTo(HaveOccurred())
+	if strings.Contains(output, "servicemonitors") {
+		isPromethuesManagedBySuite = false
+	}
+	if strings.Contains(output, "clusterserviceversions") {
+		isOLMManagedBySuite = false
+	}
+
+	if isPromethuesManagedBySuite {
+		By("Installing Prometheus")
+		Expect(tc.InstallPrometheusOperManager()).To(Succeed())
+
+		By("Ensuring provisioned Prometheus Manager Service")
+		Eventually(func() error {
+			_, err := tc.Kubectl.Get(
+				false,
+				"Service", "prometheus-operator")
+			return err
+		}, 3*time.Minute, time.Second).Should(Succeed())
+	}
+
+	if isOLMManagedBySuite {
+		By("Installing OLM")
+		Expect(tc.InstallOLM()).To(Succeed())
+	}
+
+	By("Initializing a Helm project")
+	err = tc.Init(
+		"--plugins", "helm.operator-sdk.io/v1",
+		"--project-version", "3-alpha",
+		"--domain", tc.Domain)
+	Expect(err).Should(Succeed())
+
+	By("Creating an API definition")
+	err = tc.CreateAPI(
+		"--group", tc.Group,
+		"--version", tc.Version,
+		"--kind", tc.Kind)
+	Expect(err).Should(Succeed())
+
+	By("Replacing project Dockerfile to use Helm base image with the dev tag")
+	version := strings.TrimSuffix(version.Version, "+git")
+	testutils.ReplaceInFile(filepath.Join(tc.Dir, "Dockerfile"), version, "dev")
+
+	By("Checking the kustomize setup")
+	err = tc.Make("kustomize")
+	Expect(err).Should(Succeed())
+
+	By("Building the project image")
+	err = tc.Make("docker-build", "IMG="+tc.ImageName)
+	Expect(err).Should(Succeed())
+
+	if IsRunningOnKind() {
+		By("Loading the project image into Kind cluster")
+		err = tc.LoadImageToKindCluster()
+		Expect(err).Should(Succeed())
+	}
+
+	close(done)
+}, 360)
+
+// AfterSuite run after all the specs have run, regardless of whether any tests have failed to ensures that
+// all be cleaned up
+var _ = AfterSuite(func() {
+	if isPromethuesManagedBySuite {
+		By("Uninstalling Prometheus")
+		tc.UninstallPrometheusOperManager()
+	}
+	if isOLMManagedBySuite {
+		By("Uninstalling OLM")
+		tc.UninstallOLM()
+	}
+
+	By("Destroying container image and work dir")
+	tc.Destroy()
+})
+
+// IsRunningOnKind returns true when the tests are executed in a Kind Cluster
+func IsRunningOnKind() bool {
+	return strings.Contains(Kubectx, "kind")
+}
```

Comment:
Also probably doesn't need to be exported.

```suggestion
// isRunningOnKind returns true when the tests are executed in a Kind Cluster
func isRunningOnKind() bool {
	return strings.Contains(kubectx, "kind")
}
```

Reply:
Not anymore 👍